### PR TITLE
[Dev] Dev mode for local development

### PR DESCRIPTION
**Pull request by roopakv**
# Description

If custom api, cdn and merchant dash urls are set, this allows magento2 module to work with a developer's local development setup. This basically links M2 to all of local dev.

Bonus: We can now have magento stores for staging, and during storm CD

Fixes: (link Asana Task)

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Helper/Config.php [hunk 1]
```diff
@@ -655,6 +670,15 @@ public function isSandboxModeSet($store = null)
         );
     }
 
+    public function getCustomURLValueOrDefault($path, $default)
+    {
+        $storedValue = $this->getScopeConfig()->getValue($path);
+        if ($storedValue == "") {
```
Comment (melvin0008) [inline Helper/Config.php]: return $this->getScopeConfig()->getValue($path) ?? $default

Comment (roopakv) [reply Helper/Config.php]: can't use coalesce here, the string is empty not null which coalesce doesnt check.

i can use `?:` but that will throw an E_NOTICE if not set, maybe i should be more explicit

Comment (roopakv) [reply Helper/Config.php]: i used `empty()` that should work.

**Comment by roopakv** [comment]
@matt-thomason going to merge this to unblock my next PR :) if you have any comments, def LMK and ill address them

**Pull request merged**